locksmith/src/controllers/lockController.ts

```diff
@@ -76,18 +80,30 @@ export const stripeConnected = async (req: Request, res: Response) => {
  */
 export const lockIcon = async (req: Request, res: Response) => {
   const lockAddress = Normalizer.ethereumAddress(req.params.lockAddress)
+  const network = Number(req.params.network)
   const { original } = req.query
-  const lockIcon = await getLockIcon({
+  let icon = await getLockIcon({
     lockAddress,
     original: original === '1',
     requestUrl: Normalizer.getRequestURL(req).toString(),
   })
+  if (req.query.id) {
+    // Check if there is a custom icon for this key
+    const keyIcon = await getKeyIcon({
+      network,
+      lockAddress,
+      keyId: req.query.id.toString(),
+    })
+    if (keyIcon) {
+      icon = keyIcon
+    }
+  }
```

Comment:
I know this is temporary but maybe keep it encapsulated and private within `getKeyIcon` and do the logic there?

```ts
// lockOperation.ts
export const getLockIcon = async ({
  lockAddress,
  original,
  requestUrl,
  key,
}: {
  lockAddress: string
  original?: boolean
  requestUrl: string
  key?: { id: string; network: number }
}): Promise<IconType> => {
  if (key?.id && key?.network) {
    const icon = await getCustomLockIcon({
      network: key.network,
      lockAddress,
      keyId: key.id,
    })
    if (icon) {
      return icon
    }
  }

  if (original) {
    return getGeneratedLockIcon(lockAddress)
  }

  ...
}
```

Notice this enables the `getCustomKeyIcon` to be private so the temp code doesn't get leaked into `lockController` and can even keep the `icon` variable immutable there as well. But also the `getLockIcon` logic doesn't get called unnecessarily sometimes if it knows there's a custom icon available:

```ts
// lockController.ts
const icon = await getLockIcon({
  lockAddress,
  original: original === '1',
  requestUrl: Normalizer.getRequestURL(req).toString(),
  key: {
    id: String(req.query.id || ''), // Safer then toString in case undefined
    network: parseInt(req.params.network) || 0, // Used parseInt since never expecting floats
  },
})

...
```

Reply:
I actually think it is imoportant to distinguish a `keyIcon` (individual NFT) from the `lockIcon` (icon for the contract that will be used for all keys if no custom key one is set).

Reply:
Ah I see, after straightening out keys and locks in my head, I understand the intent here better and makes perfect sense to keep them separate. Thx for explanation!